app/services/matchSurnameToAppeal.js

```diff
@@ -0,0 +1,36 @@
+const request = require('superagent');
+const { appealsAPI } = require('app/config');
+const HttpStatus = require('http-status-codes');
+
+const matchSurnameToAppeal = (req, res, next) => {
+
+  const mactoken = req.params.mactoken;
+  const surname = req.body.surname;
+
+  return request.get(`${appealsAPI}/validate/${mactoken}/${surname}`)
+    .then(result => {
+      const appealId = result.body.appealId;
+      res.redirect(`/progress/${appealId}/trackyourappeal`);
+    })
+    .catch(error => {
+      if (error.statusCode === HttpStatus.BAD_REQUEST && error.rawResponse === `Invalid surname provided: ${surname}`) {
```

Comment:
Is there a good reason why we have to check the error message? If this message changes within the API we will never fall inside the `if block` ?

Reply:
@paulgain So we need to identify what the error is so we can display the noMatch error message. I figured there could be other situations where there could be a BAD_REQUEST error that isn't to do with the surname matching the mactoken but would still display the same error instead of using the error handling pages.